### ISSUE TITLE
HeaderBackButton onPress is not firing

### ISSUE THREAD

**Issue by chalarlie** [react-navigation/react-navigation#8775]
**Current Behavior**

When setting headerLeft on a screen and overriding it with HeaderBackButton, onpress is not firing. Also, the default back behavior (when not overriding using headerLeft) is not firing. Current setup is similar to:

```
// Modal stack
<Stack.Navigator mode="modal">
      <Stack.Screen name="Main" component={TabNavigator} />
      <Stack.Screen name="ModalA" component={ModalA} />
      <Stack.Screen name="ModalA" component={ModalA} />
 />


// TabNavigator
<Tab.Navigator>
      <Tab.Screen name="HomeTab" component={Home} />
      <Tab.Screen name="SettingsTab" component={Settings} />
</Tab.Navigator>

// Settings Tab
<SettingsNavigator.Navigator headerMode="screen">
  <SettingsNavigator.Screen name="SettingsHome" component={SettingsHome} />
  <SettingsNavigator.Screen
    name="SupportedBuildings"
    component={SupportedBuildings}
    options={{
      headerShown: true,
      headerLeft: (props) => (
        <HeaderBackButton
          {...props}
          onPress={() => {
            console.log('hi');
          }}
          label="Test"
        />
      ),
    }}
  />
</SettingsNavigator.Navigator>
```

Pressing the back button does nothing and does not log to the console. Removing the "headerRight" option and using the default back button also does not correctly cause the screen to go back. HOWEVER, replacing the headerRight with a custom button as follows:

```
<SettingsNavigator.Navigator headerMode="screen">
  <SettingsNavigator.Screen name="SettingsHome" component={SettingsHome} />
  <SettingsNavigator.Screen
    name="SupportedBuildings"
    component={SupportedBuildings}
    options={{
      headerShown: true,
      headerLeft: (props) => (
        <Button
          {...props}
          onPress={() => {
            navigationRef.current?.navigate('More');
          }}
          title="Back"
        />
      )
    }}
  />
</SettingsNavigator.Navigator>
```

This WILL correctly work, which tells me it is an issue with the HeaderBackButton.

**Expected Behavior**

The onPress method for the HeaderBackButton should fire.

**How to reproduce**

Create a stack similar to the one above. Or, just try to use the HeaderBackButton. It will not work.

**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 |iOS
| @react-navigation/native       |5.7.3
| @react-navigation/stack        |5.9.0
| @react-navigation/bottom-tabs        |5.8.0
| react-native-gesture-handler   |1.7.0
| react-native-safe-area-context |3.1.6
| react-native-screens           |2.9.0
| react-native                   |0.63.0
| expo                           |none
| node                           |14.7.0
| npm or yarn                    |yarn - 1.22.4

**Comment by devproivansurzhenko** [comment]
Hey. Just found an interesting case. It is probably something similar or kind of.

I just found `headerBackImage` at my app is not touchable when click is on the image. However, it is clickable slightly above and behind.

![sample_clicks](https://user-images.githubusercontent.com/48912267/91359516-5f4f5c00-e7fd-11ea-9160-1fa372f63755.gif)

@chalarlie let me know if it is (or not) your case also. I am still researching.

**Comment by chalarlie** [comment]
Unfortunately I can't get it to fire no matter where I click :/

**Comment by devproivansurzhenko** [comment]
Yahoooo! I have found a workaround. Try yourself.

In my story:
i had `headerTitle` defined. Something like that:
```
headerTitle: () => (
        <Image
          ...
        />
    ),
```
It caused the full width expand of title. It covered the back button.
<img width="145" alt="Screenshot 2020-08-27 at 02 45 01" src="https://user-images.githubusercontent.com/48912267/91367753-f7eed780-e80f-11ea-90dc-20ab8ba32d58.png">

In some reasons, margins, paddings, zIndex are not working. I were unable to resolve it.

So, I boxed my `headerTitle` into a `View` with disabled touch events. Like the following:
```
headerTitle: () => (
      <View pointerEvents="none">
        <Image
          ...
        />
      </View>
    ),
```
In other words, `headerTitle` become transparent for touches.

It works now on my end!

**Comment by devproivansurzhenko** [comment]
@satya164 why you decided it is a question? Availability of hack does not mean that topic is a question.

**Comment by devproivansurzhenko** [comment]
@satya164 It is there:
https://github.com/devproivansurzhenko/repro-react-navigation-bug

**Comment by devproivansurzhenko** [comment]
Did you have a chance to reproduce it on your end?

**Comment by devproivansurzhenko** [comment]
@chalarlie Could you please also take a look on the repro sample?

**Comment by satya164** [comment]
https://github.com/react-navigation/react-navigation/issues/8804

**Comment by devproivansurzhenko** [comment]
@satya164 Thank you!
